webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,234 @@
+import { Button, Col, InputNumber, Modal, Row } from 'antd';
+import { Form, Input, Select } from 'antd';
+import { ModalProps } from 'antd/es/modal/Modal';
+import yaml from 'js-yaml';
+import React, { useCallback, useEffect, useState } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+
+import useStorage from 'hooks/useStorage';
+import { getResourcePools, getTemplates } from 'services/api';
+import { NotebookConfig, RawJson, ResourcePool, Template } from 'types';
+import { launchNotebook, previewNotebook } from 'utils/task';
+
+import Link from './Link';
+import RadioGroup from './RadioGroup';
+
+const { Option } = Select;
+const { Item } = Form;
+
+const STORAGE_PATH = 'notebook-launch';
+const STORAGE_KEY = 'notebook-config';
+
+interface Props extends ModalProps {
+  onLaunch?: () => void;
+  visible?: boolean;
+}
+
+const NotebookModal: React.FC<Props> = (
+  { visible = false, onLaunch = (() => null), ...props }: Props,
+) => {
+  const storage = useStorage(STORAGE_PATH);
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ resourceTypeOptions, setResourceTypeOptions ] =
+    useState<{id:string, label:string}[]>([ { id:'CPU', label:'CPU' }, { id:'GPU', label:'GPU' } ]);
+  const [ resourceType, setResourceType ] =
+    useState(storage.getWithDefault(STORAGE_KEY, { type: undefined }).type);
+  const [ form ] = Form.useForm();
+
+  useEffect(() => {
+    const fetchTemplates = async () => {
+      setTemplates(await getTemplates({}));
+    };
+    fetchTemplates();
+  }, []);
+
+  useEffect(() => {
+    const fetchResourcePools = async () => {
+      setResourcePools( await getResourcePools({}));
+    };
+    fetchResourcePools();
+  }, []);
+
+  useEffect(()=> {
+    const fetchConfig = async (values: NotebookConfig) => {
+      if(showFullConfig) {
+        const config = await previewNotebook(
+          values.slots,
+          values.template,
+          values.name,
+          values.pool,
+        );
+        form.setFieldsValue({ config: yaml.dump(config) });
+      }
+    };
+    fetchConfig(form.getFieldsValue(true));
```

Comment:
**chore:** we probably don't want to call `fetchConfig` at all if `showFullConfig` is not true. Move the `if (showFullConfig) {` to wrap around the `fetchConfig` function call.